README.adoc

```diff
@@ -226,6 +226,24 @@ make lint-install
 make format lint
 ----
 
+=== Enable tracing
+
+Kiali itself is instrumented with opentelemetry tracing to help provide insights and surface performance issues for the kiali server. To enable, set the `server.tracing_enabled` and `server.collector_url` configuration options.
+
+[source,yaml]
+----
+apiVersion: kiali.io/v1alpha1
+kind: Kiali
+metadata:
+  name: kiali
+spec:
+...
+  server:
+    collector_url: http://jaeger-collector.istio-system:14268/api/traces
+    tracing_enabled: true
```

Comment:
Let's think about this some more. Whenever I see multiple settings dealing with the same type of feature, I would like to see them grouped under the same parent heading. Otherwise, when we start adding things we start getting N related settings all spread out in the config and its confusing to users. I already saw this happen that caused me to refactor the config recently because I was adding stuff to the ingress config and it was getting confusing by spreading settings around.

We already have [metrics](https://github.com/kiali/kiali-operator/blob/v1.43.0/deploy/kiali/kiali_cr.yaml#L992-L999) settings that should also be grouped, maybe we can move those and put everything all under the same heading now.

```yaml
spec:
  server:
    observability:
      metrics:
        enabled: true
        port: 9090
      tracing:
        collector_url: http://jaeger-collector.istio-system:14268/api/traces
        enabled: true
```

Reply:
That sounds good to me. It would make it easier and more clear to add settings related to each of these settings in the future. What do you think about removing `enabled` and just having the presence of `tracing` mean that it is enabled?

```
spec:
  server:
    observability:
      tracing: {}
```
would mean tracing is enabled with only defaults. Whereas omitting tracing
```
spec:
  server:
    observability:
      metrics:
        ...
```
would mean that tracing is not enabled. One less setting to add or toggle.

Reply:
> What do you think about removing enabled

That would be against our convention. We use "enabled" for everything else. It makes it very explicit and more understandable.

Also, its easier on the implementation side - "true" or "false" is easier to deal with than "exists" or "does not exist". (think about disabling it - the user has to patch the YAML with "null" when removing settings from k8s resources and hope the merge accepts it. I know I have trouble sometimes getting the kubectl command correct when trying to patch something and REMOVE an item). It is much easier just having a boolean value.

Reply:
> That would be against our convention. We use "enabled" for everything else. It makes it very explicit and more understandable.

:+1: 

I'm generally in favor of less nesting instead of more, how about keeping `tracing` and `metrics` under the `server` header?
```
spec:
  server:
      metrics:
        enabled: true
        port: 9090
      tracing:
        collector_url: http://jaeger-collector.istio-system:14268/api/traces
        enabled: true
```

Reply:
Also does this change need to be backwards compatible? i.e. should we keep the `metrics_enabled` field and deprecate it to be removed at a later date or no?

Reply:
> how about keeping tracing and metrics under the server header?

On second thought, if the otel integration is moving under an `observability` package, it makes sense to have this as part of the config.

Reply:
> Also does this change need to be backwards compatible? i.e. should we keep the metrics_enabled field and deprecate it to be removed at a later date or no?

Right now - unless I can think of a reason otherwise - I am thinking don't worry about it. The only reason why I had to add support for the backward-compatible ingress setting was because the Maistra/OSSM operator was setting that deprecated setting, so I wanted to maintain support for it. But OSSM operator (AFAIK) does NOT set this server.metrics_enabled setting - so I don't see a need to maintain backward compatibility for it. It would be easy to add if it turns out there is a reason for it, but as of right now I don't see a need to add that technical debt to the code. This is a minor feature that even if someone missed it by using the old setting, nothing horrible will happen and its easily fixed on the user's end.

Reply:
After the refactoring, this will require some operator changes. I'll push those shortly.